leo-finalize.md

```diff
@@ -0,0 +1,80 @@
+
+### Leo Finalize Update
+
+The finalize programming model for on-chain code has been replaced with a new async/await model.
+
+The key difference between the two models is that instead of a finalize block, on-chain code is defined in an async function. Note that if a program does not declare on-chain code (there are not finalize blocks) then it will work without modification under the async/await model.
+
+### Previous Testnet3 Leo version Finalize model
+program test1.aleo {
+
+    //sample mapping data
+    mapping data: u32 => u32; 
+
+    //transition
+    transition main(public x: u32) -> (u32) {
+        //finalize `main` called in the finalize state below, must match with transition name
+        return (x) then finalize(x);
+    }
+
+    finalize main (x: u32){
+    //add up x value to x_1
+    let x_1: u32 = x + 2u32;
+    //add up x value to x_2
+    let x_2: u32 = x + 5u32;
+    //conditional statement check if x_1 is greater than x_2
+    //if true, output x_1 otherwise x_2 to mapping (data)
+    x = (x_1 > x_2) ? x_1 : x_2;
+    Mapping::set(data, 0u32, x);
+    } }
+    
+### Testnet Beta Leo version Finalize model
+program test1.aleo {
+
+    //sample mapping data
+    mapping data: u32 => u32; 
+
+    //async transition
+    async transition main(public x: u32) -> (u32, Future) {
+        //function `finalize_main` called in the async state below
+        return (x, finalize_main(x));
+    }
+
+    async function finalize_main (x: u32){
+    //add up x value to x_1
+    let x_1: u32 = x + 2u32;
+    //add up x value to x_2
+    let x_2: u32 = x + 5u32;
+    //conditional statement check if x_1 is greater than x_2
+    //if true, output x_1 otherwise x_2 to mapping (data)
+    x = (x_1 > x_2) ? x_1 : x_2;
+    Mapping::set(data, 0u32, x);
+    } }
+
+
+### Key Concepts
+1. Async Functions and Transitions:
+
+- Async functions can be defined to run asynchronously. These functions return a Future, which represents code that will be executed later.
+- Async transitions are special kinds of async functions that can be called to perform state changes on the blockchain.
+
+2. Futures:
+
+- A Future is an object representing a value that will be available at some point in the future. In the async/await model, functions do not return values directly; instead, they return Futures.
+
+### Rules of Async/Await Model
```

Comment:
I think it could be more correct to refer to the Async/Await update as a "syntax" instead of as a "model". 
In general this syntax support an asynchronous programming pattern. But calling it a model could be an over reach.